python/unityagents/environment.py

```diff
@@ -457,6 +457,11 @@ def _get_state(self, output: UnityRLOutput) -> (AllBrainInfo, bool):
             else:
                 [x.memories.extend([0] * (memory_size - len(x.memories))) for x in agent_info_list]
                 memory = np.array([x.memories for x in agent_info_list])
+            m_actions = np.ones((len(agent_info_list), self.brains[b].vector_action_space_size))
+            for i, x in enumerate(agent_info_list):
+                if x.masked_actions is not None:
+                    for k in range(len(x.masked_actions)):
```

Comment:
- should check for length compatibility and throw a helpful error msg if invalid
- can we use array ops to remove the loop?